sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/BeamFnControlClient.java

```diff
@@ -159,6 +159,16 @@ public void processInstructionRequests(Executor executor)
             } catch (Error e) {
               sendErrorResponse(e);
               throw e;
+            } catch (IllegalStateException e) {
```

Comment:
Do we only care about `IllegalStateException ` here? It seems like other exceptions might also be thrown.

Reply:
Instead of adding this try catch, we should up date the handler to not throw the exception in the first place.
Here: https://github.com/apache/beam/blob/bca893e86a04f7245df1476597f157b3b6fa0651/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java#L378
Here: https://github.com/apache/beam/blob/bca893e86a04f7245df1476597f157b3b6fa0651/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java#L344

We should add a simple unit test that covers this as well.

Reply:
I don't think that will resolve the issue of runner waiting for a response here and timing out (with a bad error in the case of Dataflow).

How about catching all exceptions here and sending a error response to the runner set here: https://github.com/apache/beam/blob/master/model/fn-execution/src/main/proto/beam_fn_api.proto#L129 

?

Reply:
Nvm, we already do it here: https://github.com/apache/beam/blob/master/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/BeamFnControlClient.java#L182

I'm looking more closely at this now. Sorry about the delay.

Reply:
I would suggest sending back an empty successful response instead of throwing an error. Empty progress/split responses are no-ops from the runners side.

Reply:
It took me forever to get to this PR so I wouldn't worry about the delay.

Reply:
Updated to send an empty successful response instead of failing. Also added unit tests.